src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleDictionary.cs

```diff
@@ -30,28 +30,9 @@ private static void HandleStartDictionary(JsonSerializerOptions options, ref Rea
                 state.Current.InitializeJsonPropertyInfo();
                 state.Current.CollectionPropertyInitialized = true;
 
-                ClassType classType = state.Current.JsonClassInfo.ClassType;
```

Comment:
Can you elaborate on this code removal change (and the one below) and why it's covered elsewhere? Thanks

Reply:
For the `ClassType.Value` check, the `true` case was unreachable because we check if we are processing `ClassType.Value` here: https://github.com/dotnet/corefx/blob/4c3bd8a0533a2e35df371b91045a1a5f0ada7d54/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs#L63-L70

For the `ClassType.Enumerable` check, processing a nested enumerable (after we have pushed the element type to the read stack) means that we have just seen a StartObject token instead of a StartArray token, which in turn means we have bad/mismatched Json (pending semantics/implementation of Ref Handling). Thus, we should thow `JsonException` for this state.

Here's the coverage for current master: https://ibb.co/mhgH1TT
For the reasons given above, the ClassType.Value check is never hit, and processing ClassType.Enumerable here will always lead to a `JsonException` (CreateObject is always null because for enumerables, it can only be set after we see a StartArray token.)

Reply:
I see. Thanks for the detail. When `IsProcessingValue()` was added this became dead code (I assume).